### DISCUSSION TITLE
Any chance to loosen requirements on httpx?

### DISCUSSION THREAD

**Discussion by r-xyz** [python-telegram-bot/python-telegram-bot#4147] [Feature Suggestions]
Hello,
First of all thanks for the great library.
I am opening the topic only as a concerned user :)
Some Home Assistant devs [are trying to keep the library up-to-date](https://github.com/home-assistant/core/pull/110297) inside [homeassistant/core](https://github.com/home-assistant/core).
Unfortunately, due to current pinning strategy of httpx (`=~0.X.0`) and mismatching release schedules, at the moment it seems difficult to have the library updated. (Home Assistant is usually faster in implementing `httpx` updates and this results in dependencies conflicts).
Leaving current version frozen means not only missing new features in Home Assistant, but also having many previously implemented features which are getting broken (current version is 13.1 from November 2020).
I was wondering if there is any chance to have httpx requirement loosened to include higher versions, or if you could suggest any other solutions.
Thanks in advance :)

**Comment by Bibo-Joshi** [comment]
Thanks for reaching out.

We have used a relatively strict pinning for httpx so far b/c it does not have a stable release yet. However, we have had no stability issues with httpx at all since adopting it in mid 2022 even though there have been a number of httpx release since then.

As such I would be open to adjusting the bound upper bound to `~=0.27`, allowing `~0.X.Y >= 0.27.0` as well, but not `1.0.0`. Relaxing the lower bound is more difficult, as there have been some deprecations and new features that we rely on, so that I'd like to keep 0.27 as minimal required version.
@harshil21 @clot27 @Poolitzer would you agree?

On a side note: httpx v1.0.0 does seem to be getting closer, see https://github.com/encode/httpx/pull/3069

**Reply by Poolitzer** [reply]
I dont mind this

**Reply by r-xyz** [reply]
Thanks a lot for taking it into consideration! :)

**Reply by harshil21** [reply]
agree, should not have consequences.

**Reply by clot27** [reply]
+1 from my side

**Reply by r-xyz** [reply]
Great, thanks a lot! :) I close the discussion for now since it is solved by #4148.

**Reply by ScotterMonk** [reply]
After this version of telegram bot, it wants 0.27.0 of httpx, which is incompatible with openai 1.30.4 that will use up to 0.26.0 of httpx. Ideas?

**Reply by Bibo-Joshi** [reply]
According to https://github.com/openai/openai-python/blob/811f4e736b75a418fb9bc579861971b239e6cd0c/pyproject.toml#L11, the requirement of openai should be compatible with ours
